Lombiq.Hosting.Tenants.Management/Service/SetupWithRecipesFilterService.cs

```diff
@@ -18,60 +12,36 @@ namespace Lombiq.Hosting.Tenants.Management.Service
-        private readonly IOptions<HideRecipesFromSetupOptions> _hideRecipesFromSetupOptionsOptions;
-        private readonly IOptions<AutoSetupOptions> _autoSetupOptionsOptions;
+        private readonly IOptions<HideRecipesFromSetupOptions> _hideRecipesFromSetupOptions;
         private readonly ShellSettings _shellSettings;
+        private readonly ISetupService _setupService;
 
-#pragma warning disable S107 // Methods should not have too many parameters
         public SetupWithRecipesFilterService(
-            IShellHost shellHost,
-            IHostEnvironment hostEnvironment,
-            IShellContextFactory shellContextFactory,
-            ISetupUserIdGenerator setupUserIdGenerator,
-            IEnumerable<IRecipeHarvester> recipeHarvesters,
-            ILogger<SetupService> logger,
-            IStringLocalizer<SetupService> stringLocalizer,
-            IHostApplicationLifetime hostApplicationLifetime,
-            IHttpContextAccessor hca,
-            IOptions<HideRecipesFromSetupOptions> hideRecipesFromSetupOptionsOptions,
-            IOptions<AutoSetupOptions> autoSetupOptionsOptions,
-            ShellSettings shellSettings)
-#pragma warning restore S107 // Methods should not have too many parameters
+            IOptions<HideRecipesFromSetupOptions> hideRecipesFromSetupOptions,
+            ShellSettings shellSettings,
+            ISetupService setupService)
         {
-            _setupService = new(
-                shellHost,
-                hostEnvironment,
-                shellContextFactory,
-                setupUserIdGenerator,
-                recipeHarvesters,
-                logger,
-                stringLocalizer,
-                hostApplicationLifetime,
-                hca);
-
-            _hideRecipesFromSetupOptionsOptions = hideRecipesFromSetupOptionsOptions;
-            _autoSetupOptionsOptions = autoSetupOptionsOptions;
+            _hideRecipesFromSetupOptions = hideRecipesFromSetupOptions;
             _shellSettings = shellSettings;
+            _setupService = setupService;
         }
 
         public async Task<IEnumerable<RecipeDescriptor>> GetSetupRecipesAsync()
         {
             var recipesDescriptors = await _setupService.GetSetupRecipesAsync();
-            var currentTenantSetupOptions = _autoSetupOptionsOptions.Value.Tenants.FirstOrDefault(tenant
-                => tenant.ShellName == _shellSettings.Name);
 
-            // The first case is when specify the tenant recepe name via default tenant admin UI,
-            // the second case is when the tenant is already configured via AutoSetup feature.
-            if (_shellSettings["RecipeName"] != null || currentTenantSetupOptions?.RecipeName != null)
+            // The first case is when specify the tenant recipe name via the Default tenant admin UI or AutoSetup
+            // feature, the second case is necessary because the default tenant doesn't fill in RecepeName even if we
+            // use auto setup.
```

Comment:
Ez a "recepe" egy visszatérő becsípődés :).

```suggestion
            // The first case is when we specify the tenant recipe name via the Default tenant admin UI or AutoSetup
            // feature, the second case is necessary because the Default tenant doesn't fill in RecipeName even if we
            // use auto setup.
```